Sources/Datadog/Traces/DDSpan.swift

```diff
@@ -0,0 +1,89 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import OpenTracing
+import Foundation
+
+internal class DDSpan: Span {
+    private(set) var operationName: String
+    /// The `Tracer` which created this span.
+    internal let issuingTracer: DDTracer
+    internal let startTime: Date
+    internal var isFinished: Bool = false
+
+    init(tracer: DDTracer, operationName: String, parentSpanContext: DDSpanContext?, startTime: Date) {
+        self.issuingTracer = tracer
+        self.operationName = operationName
+        self.startTime = startTime
+        self.context = DDSpanContext(
+            traceID: parentSpanContext?.traceID ?? .generateUnique(),
+            spanID: .generateUnique()
+        )
+    }
+
+    // MARK: - Open Tracing interface
+
+    let context: SpanContext
+
+    func tracer() -> Tracer {
+        return issuingTracer
+    }
+
+    func setOperationName(_ operationName: String) {
+        guard warnIfFinished("setOperationName(_:)") else {
+            return
+        }
+        self.operationName = operationName
+    }
+
+    func setTag(key: String, value: Codable) {
+        guard warnIfFinished("setTag(key:value:)") else {
+            return
+        }
+        // TODO: RUMM-292
+    }
+
+    func setBaggageItem(key: String, value: String) {
+        guard warnIfFinished("setBaggageItem(key:value:)") else {
+            return
+        }
+        // TODO: RUMM-292
+    }
+
+    func baggageItem(withKey key: String) -> String? {
+        guard warnIfFinished("baggageItem(withKey:)") else {
+            return nil
+        }
+        // TODO: RUMM-292
+        return nil
+    }
+
+    func finish(at time: Date) {
```

Comment:
Shouldn't there be a default implementation of `finish` that gets the current time automatically ? That's the way it's implemented on our side, users can simply write `span.finish()`

Reply:
They define `func finish()` in `opentracing-swift`, [here](https://github.com/opentracing/opentracing-swift/blob/e2b0551de2fd9a5cd9a3d8dff5390440f724283c/Sources/OpenTracing/Span.swift#L57).

Thanks to Swift language specificies they do it as `Span` interface extension:
```swift
public extension Span {
   // ...

   /// Finish the span at the current time
   func finish() {
       self.finish(at: Date())
   }
}
```